### PR TITLE
[WIP] moves Sufia models which extend Legato::Model into an initializer to …

### LINKED ISSUE #2380: undefined method 'sufia__pageview' while rendering stats page
In non-production environments, rendering the stats view for a work results in : `NoMethodError: undefined method `sufia__pageview' for #<Legato::Management::Profile:0x007f1c449cc1f8>
...`

We found that the PageView and Download models weren't being eager loaded in non-production environments, and the fix was very simple yet not very clear or easy to identify at first.

I'm proposing we introduce a more clear/simple configuration or initializer for models which extend the `Legato::Model` and get the dynamic `sufia__???????` method injected for all application environments.

I'll put together a PR and we can adjust the implementation as necessary.

### PR DESCRIPTION
**Pull request by revgum**
Fixes #2380

### CODE REVIEW

**Comment by awead** [comment]
So this issue affects you if you're in development and test, but not in production?

**Comment by mjgiarlo** [comment]
@revgum :arrow_up: :question:

**Comment by mjgiarlo** [comment]
@awead I'm surprised y'all haven't run into this before with ScholarSphere.

**Comment by revgum** [comment]
@awead @mjgiarlo Yea, we observed this in development and "staging" environments.

File: lib/generators/sufia/templates/config/legato.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# If Google Analytics are enabled, pre-load the models which rely on
+# Legato::Model.extended(base) dynamic profile method generation. Non-production
+# environments wouldn't necessarily have this eager loaded by default.
+if Sufia.config.analytics
```
Comment (awead) [inline lib/generators/sufia/templates/config/legato.rb]: If this doesn't affect production, can you add ` && Rails.env != "production"` or something equivalent?

**Comment by jcoyne** [comment]
Please hold off on merging this until the bug can be independently replicated. I attempted to do so, but was unable to.

**Comment by revgum** [comment]
This is somewhat troubling.. I tested this manually using Sufia 7.0.0 engine cart generated app and it works fine, but when we use Sufia 7.0.0 gem in our instance, we get the same undefined method error. 

```
NoMethodError (undefined method `sufia__pageview' for #<Legato::Management::Profile:0x007faa26113230>):
  /Users/joshgum/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/sufia-850b9062772e/app/models/sufia/statistic.rb:35:in `ga_statistics'
  /Users/joshgum/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/sufia-850b9062772e/app/models/sufia/statistic.rb:50:in `combined_stats'
  /Users/joshgum/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/sufia-850b9062772e/app/models/sufia/statistic.rb:28:in `statistics'
  /Users/joshgum/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/sufia-850b9062772e/app/presenters/work_usage.rb:15:in `initialize'
  /Users/joshgum/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/sufia-850b9062772e/app/controllers/stats_controller.rb:8:in `new'
  /Users/joshgum/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/sufia-850b9062772e/app/controllers/stats_controller.rb:8:in `work'
  actionpack (4.2.6) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
...
```

**Comment by mjgiarlo** [comment]
@revgum In which environment(s) are you seeing this now?

**Comment by revgum** [comment]
@mjgiarlo I was using "development" in the most recent testing that failed with the sufia__pageview NoMethodError

**Comment by mjgiarlo** [comment]
Thx, @revgum. Regarding @jcoyne's message above, it would be good if someone could verify/reproduce this before we merge the PR.

**Comment by revgum** [comment]
@mjgiarlo Agreed. I wonder if `engine_cart:generate` is doing something more/different than `rails generate sufia:install -f` when installing sufia on a fresh rails app? This all seems strange to me.. I'm happy to keep this wedge in place in ScholarsArchive, but figured it would be worthwhile making its way into Sufia unless we somehow missed something on our end..

**Comment by mjgiarlo** [comment]
@revgum The Sufia test app *does* do more than run the install generator, but nothing here looks particularly suspicious to me: https://github.com/projecthydra/sufia/blob/master/spec/test_app_templates/lib/generators/test_app_generator.rb

I'm happy to do some legwork on this one if you can help me figure out how to see what you're seeing on my side. Is it possible that you've otherwise configured your app not to do eager loading in the dev env? Are you seeing the weird behavior on your laptops or in a server environment? Is it occurring on multiple hosts? How would I reproduce this? Pull down ScholarsArchive, toss an Analytics config file in there, and then :monkey: around in the Rails console?

@cam156 @awead do y'all have any thoughts about this? Are analytics working A-OK in your ScholarSphere-atop-Sufia-7 testing?

**Comment by cam156** [comment]
@mjgiarlo the analytics are not working in ScholarSphere, but we have other issues https://github.com/psu-stewardship/scholarsphere/issues/345

**Comment by mjgiarlo** [comment]
@cam156 Boo. :unamused:

**Comment by mjgiarlo** [comment]
@revgum I wrote this up as an issue in #2460. When we get someone to replicate the issue, we can look into applying the patch you pushed in your branch. Closing this PR for now. Thanks!

**Pull request closed without merging**